src/odemis/odemisd/test/modelgen_test.py

```diff
@@ -0,0 +1,165 @@
+
+        # __init__ contains a setting which is overwritten of the CL-detector
+        with open("yaml-merger-include-complete-overwrite-init-CL-Detector-test.odm.yaml", "r") as f:
+            data_found = yaml.load(f, SafeLoader)
+
+        # Compare with expected results
+        expected_result = self.expected_full_result["CL Detector"]["init"].copy()  # Copy because a value is adjusted
+        expected_result["settle_time"] = ["overwritten", "values", "are", "correctly", "stored"]
+        self.assertEqual(expected_result, data_found)
+
+        # __init__ of the CL-detector using a relative reference in the !include
+        with open(
+            "yaml-merger-relative-path-test/yaml-merger-include-complete-init-relative-path-CL-Detector-test.odm.yaml",
+                "r") as f:
+            data_found = yaml.load(f, SafeLoader)
+
+        # Compare with expected results
+        expected_result = self.expected_full_result["CL Detector"]["init"]
+        self.assertEqual(expected_result, data_found)
+
+        # __init__ of the CL-detector using an absolute reference in the !include
+        with open(
+            "yaml-merger-relative-path-test/yaml-merger-include-complete-init-relative-path-CL-Detector-test.odm.yaml",
+                "r") as f:
+            data_found = yaml.load(f, SafeLoader)
+
+        # Compare with expected results
+        expected_result = self.expected_full_result["CL Detector"]["init"]
+        self.assertEqual(expected_result, data_found)
+
+        with self.assertRaises(FileNotFoundError):
+            with open("yaml-merger-include-error-non-exist-file-in-init-CL-Detector-test.odm.yaml", 'r') as f:
+                data_found = yaml.load(f, SafeLoader)
+
+        with self.assertRaises(ParseError):
+            with open("yaml-merger-include-error-reference-to-typo-in-init-CL-Detector-test.odm.yaml", 'r') as f:
+                data_found = yaml.load(f, SafeLoader)
+
+    def test_extend(self):
+        """
+        Basic test for the extend keyword in a yaml file to extend a dictionary with the content of a file.
+        """
+        # __init__ of the SEM Scan Interface
+        with open("yaml-merger-extend-complete-init-SEM-Scan-Interface-test.odm.yaml", "r") as f:
+            data_found = yaml.load(f, SafeLoader)
+
+        # Compare with expected results
+        expected_result = {"init": self.expected_full_result["SEM Scan Interface"]["init"]}
+        self.assertEqual(expected_result, data_found)
+
+        # Full SEM Scan Interface component
+        with open("yaml-merger-extend-full-SEM-Scan-Interface-component-test.odm.yaml", "r") as f:
+            data_found = yaml.load(f, SafeLoader)
+
+        # Compare with expected results
+        expected_result = {"SEM Scan Interface": self.expected_full_result["SEM Scan Interface"]}
+        self.assertEqual(expected_result, data_found)
+
+        # __init__ contains a entries which are overwritten of the CL-detector
```

Comment:
typo: "contains an entry" or "contains entries"
And some comment as above, I think you mean "overwritten by"